code/game/machinery/buttons.dm

```diff
@@ -84,11 +84,23 @@
 /obj/machinery/driver_button/multitool_menu(var/mob/user, var/obj/item/multitool/P)
 	return {"
 	<ul>
-	<li><b>ID Tag:</b> [format_tag("ID Tag","id_tag")]</li>
+	<li><b>ID Tag:</b> [format_tag("ID Tag","id_tag","set_id")]</li>
 	<li><b>Logic Connection:</b> <a href='?src=[UID()];toggle_logic=1'>[logic_connect ? "On" : "Off"]</a></li>
 	<li><b>Logic ID Tag:</b> [format_tag("Logic ID Tag", "logic_id_tag")]</li>
 	</ul>"}
 
+/obj/machinery/driver_button/multitool_topic(var/mob/user, var/list/href_list, var/obj/O)
```

Comment:
So this this proc you've written is very similar to the base machinery one:
```dm
/obj/machinery/proc/multitool_topic(mob/user, list/href_list, obj/O)
	if("set_id" in href_list)
		if(!("id_tag" in vars))
			warning("set_id: [type] has no id_tag var.")
		var/newid = copytext(reject_bad_text(input(usr, "Specify the new ID tag for this machine", src, src:id_tag) as null|text),1,MAX_MESSAGE_LEN)
		if(newid)
			src:id_tag = newid
			return TRUE
```
It does have the cursed `:` format but still, it looks like it does everything your new proc does. Do you know why the base proc doesn't work for the button and driver?

Reply:
Turns out I'm an idiot.
You're absolutely right, current base machinery proc works just fine, I'd just wrongly assumed it didn't work on the mass driver stuff for whatever reason. All that was missing was the set_id format tags. Should've probably tested just that first.